### PR TITLE
Update Use cases

### PR DESCRIPTION
**Pull request by Jh123x**
Linked to #250 

Add Use case for:
- [x] Import command
- [x] Export command

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -508,7 +509,8 @@ Use case ends.
 **Extensions**
 - 1a. The format of add method is invalid
     1a1. McGymmy shows an error message
-    Use case ends.
+    
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: I think this line should be tabbed in, based on this example in the notebook
image

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -523,64 +525,213 @@ Use case ends
 **Extensions**
 - 2a. The list is empty<br>
 
-    Use case ends.
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Same for this

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 
 - 3a. The given index is invalid.<br>
    - 3a1. McGymmy shows an error message.
 
-    Use case resumes at step 2.
+Use case resumes at step 2.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: And this

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+ - 1a. The macro does not exist
+    - 1a1. McGymmy shows the error message
 
 Use case ends.
 
+ - 2a. One of the executed commands encounter an error.
+    - 2a1. McGymmy shows the error message from that command, and displays the commands that successfully executed, and the commands that have yet to execute.
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:726]: Should be tabbed in

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+2. McGymmy executes the commands in the macro command as if the user entered them one by one
+
+Use case ends
+
+**Extensions**
+
+ - 1a. The macro does not exist
+    - 1a1. McGymmy shows the error message
 
 Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Should be tabbed in

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+2. McGymmy adds the macro to the list of available commands
+
+Use case ends
+
+**Extensions**
+
+ - 1a. The format of the macro is invalid
+    - 1a1. McGymmy shows an error message
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Should be tabbed in

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+1. User creates a macro to execute several commands in sequence
+2. McGymmy adds the macro to the list of available commands
+
+Use case ends
+
+**Extensions**
+
+ - 1a. The format of the macro is invalid
+    - 1a1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of these lines

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+    - 1b1. McGymmy shows an error message.
+
+Use case ends.
+    
+**Use case: UC11 Export a McGymmy data file**
+
+**MSS**
+
+1. User exports his McGymmy file
+2. McGymmy creates a copy of his MyGymmy file in the selected directory
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of these lines

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+Use case ends.
+
+**Extensions**
+ - 1a. The format for the save file is invalid.
+    - 1a1. McGymmy shows an error message.
+ - 1b. The path to the save file is invalid.
+    - 1b1. McGymmy shows an error message.
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+**MSS**
+
+1. User imports McGymmy file to McGymmy.
+2. McGymmy overrides the old files with the new McGymmy files.
+
+Use case ends.
+
+**Extensions**
+ - 1a. The format for the save file is invalid.
+    - 1a1. McGymmy shows an error message.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing "Use case ends." for extension 1a

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+1. User filters list (UC06)
+2. User requests to clear the displayed list
+3. McGymmy deletes all food items in the current displayed list
+
+Use case ends.
+
+**Extensions**
+- 1a. The list is empty<br>
+    
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
- - 1a. The format of the macro is invalid.
+**MSS**
+1. User filters list (UC06)
+2. User requests to clear the displayed list
+3. McGymmy deletes all food items in the current displayed list
+
+Use case ends.
+
+**Extensions**
+- 1a. The list is empty<br>
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the sentence

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+Use case ends.
+    
+**Use case: UC09 Clear**
 
- - 1a. The format of the macro is invalid.
+**MSS**
+1. User filters list (UC06)
+2. User requests to clear the displayed list
+3. McGymmy deletes all food items in the current displayed list
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of the sentence

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+1. User requests to find (filter) food items based on certain criteria
+2. McGymmy shows an updated list based on the filter criteria
+
+Use case ends.
+
+**Extensions**
+- 1a. The format of find method is invalid
+    1a1. McGymmy shows an error message
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of the sentence, use case ends should be tabbed in

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+- 3c. The format of untag method is invalid
+    3c1. McGymmy shows an error message
+
+Use case ends.
+    
+**Use case: UC08 Find**
+
+**MSS**
+1. User requests to find (filter) food items based on certain criteria
+2. McGymmy shows an updated list based on the filter criteria
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of the lines

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+- 3b. The given tag does not exist for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+   
+Use case resumes at step 2.
+
+- 3c. The format of untag method is invalid
+    3c1. McGymmy shows an error message
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
+
+- 3b. The given tag does not exist for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+   
+Use case resumes at step 2.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 
 **Extensions**
+- 2a. The list is empty<br>
+
+Use case ends.
+
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+2. McGymmy shows a list of food
+3. User request to remove tag from a specific food on the list
+4. McGymmy removes the tag from the food
 
 Use case ends
 
 **Extensions**
+- 2a. The list is empty<br>
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 20]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+Use case resumes at step 2.
+
+- 3b. The given tag does not exist for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+   
+Use case resumes at step 2.
+
+- 3c. The format of untag method is invalid
+    3c1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 
 Use case ends
 
 **Extensions**
+- 2a. The list is empty<br>
+
+Use case ends.
+
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 22]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+**MSS**
+1. User requests to list food (UC01)
+2. McGymmy shows a list of food
+3. User request to remove tag from a specific food on the list
+4. McGymmy removes the tag from the food
 
 Use case ends
 
 **Extensions**
+- 2a. The list is empty<br>
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 23]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+Use case ends.
+    
+**Use case: UC07 Untag food**
+
+**MSS**
+1. User requests to list food (UC01)
+2. McGymmy shows a list of food
+3. User request to remove tag from a specific food on the list
+4. McGymmy removes the tag from the food
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 24]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+- 3b. The given tag already exists for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
+
+- 3c. The format of tag method is invalid
+    3c1. McGymmy shows an error message
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 25]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
+Use case resumes at step 2.
+
+- 3b. The given tag already exists for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
+
+- 3c. The format of tag method is invalid
+    3c1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 26]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+    
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
+
+- 3b. The given tag already exists for the selected food.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 27]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+    
+Use case ends.
+    
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
+
+Use case resumes at step 2.
+
+- 3b. The given tag already exists for the selected food.<br>
+   - 3a1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 28]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
-4. McGymmy executes the two commands consecutively
+Use case ends
+
+**Extensions**
+- 2a. The list is empty<br>
+    
+Use case ends.
+    
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 29]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+4. McGymmy updates the food with the given tag
 
-1. User creates a macro to execute two 'add' commands in sequence
-2. McGymmy adds the macro to the list of available commands
-3. User uses the newly added macro command
-4. McGymmy executes the two commands consecutively
+Use case ends
+
+**Extensions**
+- 2a. The list is empty<br>
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 30]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
-**Use case: UC06 Add a macro command**
+Use case ends.
+    
+**Use case: UC06 Tag food**
 
 **MSS**
+1. User requests to list food (UC01)
+2. McGymmy shows a list of food
+3. User request to tag a specific food on the list
+4. McGymmy updates the food with the given tag
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of the lines

File: docs/DeveloperGuide.md [hunk 31]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
    - 3a1. McGymmy shows an error message
-    Use case resumes at step 2.
+    
+Use case resumes at step 2.
+
+- 3b. The format of edit method is invalid
+    3b1. McGymmy shows an error message
 
-**Use case: UC06 Add a macro command**
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 32]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 - 2a. The list is empty<br>
-    Use case ends.
+    
+Use case ends.
+
 - 3a. The given index is invalid.<br>
    - 3a1. McGymmy shows an error message
-    Use case resumes at step 2.
+    
+Use case resumes at step 2.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 33]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+3. User request to edit a specific food on the list
 4. McGymmy updates the food
 
 Use case ends
 
 **Extensions**
 - 2a. The list is empty<br>
-    Use case ends.
+    
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:575]: This should be tabbed in

File: docs/DeveloperGuide.md [hunk 34]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
+
 - 3a. The given index is invalid.<br>
    - 3a1. McGymmy shows an error message
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 35]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 1. User requests to list food (UC01)
 2. McGymmy shows a list of food
-3. User request to update a specific food on the list
+3. User request to edit a specific food on the list
 4. McGymmy updates the food
 
 Use case ends
 
 **Extensions**
 - 2a. The list is empty<br>
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:574]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 36]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 
-**Use case: UC05 Update food**
+**Use case: UC05 Edit food**
 
 **MSS**
 1. User requests to list food (UC01)
 2. McGymmy shows a list of food
-3. User request to update a specific food on the list
+3. User request to edit a specific food on the list
 4. McGymmy updates the food
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dot at the end of the line

File: docs/DeveloperGuide.md [hunk 37]
```diff
@@ -523,64 +525,213 @@ Use case ends
...
 
 **MSS**
 1. User requests help
-2. McGymmy shows all commands and examples of command usages
+2. McGymmy shows all commands and the usage of each command
 
 Use case ends
 
 **Extensions**
 No extensions
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Missing the dots at the end of the line

**Review by aidoxe-123 — CHANGES_REQUESTED** [review]
The content LGTM. You should keep some "Use case ends" tabbed in (based on an example from the textbook), and have consistent usage of dots at the end of the line.
image

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (chewypiano) [reply docs/DeveloperGuide.md]: I dont think the example needs to followed religiously, i believe just having it standardised is sufficient

File: docs/DeveloperGuide.md [hunk 38]
```diff
@@ -493,17 +493,15 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 Use case ends.
 
 **Use case: UC02 Add food**
 
 **MSS**
 1. User requests to add food into the list.
-2. McGymmy adds the food item into the list.
-
+2. McGymmy adds the food item into the list.<br>
 Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:524]: Missing break line
image

File: docs/DeveloperGuide.md [hunk 39]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
 
 **MSS**
 1. User requests help.
-2. McGymmy shows all commands and examples of command usages.
-
+2. McGymmy shows all commands, and the usage of each command.<br>
 Use case ends.
 
 **Extensions**
 No extensions.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Not sure if we should have a break line before this

File: docs/DeveloperGuide.md [hunk 40]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
+   - 3c1. McGymmy shows an error message.<br>
+   Use case ends.
+    
+**Use case: UC08 Find**
+
+**MSS**
+1. User requests to find (filter) food items based on certain criteria.
+2. McGymmy shows an updated list based on the filter criteria.
+
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:640]: Should have no break line before this line

File: docs/DeveloperGuide.md [hunk 41]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
+
+**Extensions**
+- 1a. The format of find method is invalid.
+   - 1a1. McGymmy shows an error message.
 
 Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Should have no break line before this line

File: docs/DeveloperGuide.md [hunk 42]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
+**Use case: UC09 Clear**
 
- - 4a. One of the executed commands encounter an error.
-    - 4a1. McGymmy shows the error message from that command, and displays the commands that successfully executed, and the commands that have yet to execute.
+**MSS**
+1. User filters list (UC06).
+2. User requests to clear the displayed list.
+3. McGymmy deletes all food items in the current displayed list.
 
 Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:653]: Should have no break line before this line

File: docs/DeveloperGuide.md [hunk 43]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
 
-**Use case: UC07 Undo the last command**
+**Extensions**
+- 1a. The list is empty.<br>
+    
+Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: Should have no break line before this line

File: docs/DeveloperGuide.md [hunk 44]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
-1. User requests to undo the last command.
-2. McGymmy successfully undo the last command.
+1. User imports McGymmy file to McGymmy.
+2. McGymmy overrides the old files with the new McGymmy files.<br>
+Use case ends.
 
+**Extensions**
+ - 1a. The format for the save file is invalid.
+    - 1a1. McGymmy shows an error message.
+    Use case ends.
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md:670]: missing break line before this line
image

File: docs/DeveloperGuide.md [hunk 45]
```diff
@@ -527,76 +525,200 @@ Use case ends
...
+
+**MSS**
 
+1. User requests to undo the last command.
+2. McGymmy successfully undo the last command.<br>
 Use case ends.
 
+**Extensions**
+
+1a. McGymmy shows the error message that there is no command left to undo.<br>
```
Comment (aidoxe-123) [inline docs/DeveloperGuide.md]: missing a `-` before 1a

**Review by aidoxe-123 — CHANGES_REQUESTED** [review]
Almost LGTM. Still have some formatting errors left

File: docs/DeveloperGuide.md (hunk 39 above)
Comment (Jh123x) [reply docs/DeveloperGuide.md]: No break will result in them being on the same line

Comment (Jh123x) [reply docs/DeveloperGuide.md]: Actually u can try to view in the rich text version and see

File: docs/DeveloperGuide.md (hunk 40 above)
Comment (Jh123x) [reply docs/DeveloperGuide.md:640]: Same thing here

File: docs/DeveloperGuide.md (hunk 39 above)
Comment (Jh123x) [reply docs/DeveloperGuide.md]: Wait nvm edited

**Review by aidoxe-123 — APPROVED** [review]
LGTM, except I'm not sure if we should separate the extension of UC04 into 2 lines. Can merge alr if we decide to keep it that way
image

**Comment by Jh123x** [comment]
> LGTM, except I'm not sure if we should separate the extension of UC04 into 2 lines. Can merge alr if we decide to keep it that way
> image

Good catch. Will edit that

**Review by aidoxe-123 — APPROVED** [review]
LGTM!

**Pull request merged**